File: Domain/Entities/Category.cs
using System;
using System.Collections;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Text.Json.Serialization;

namespace Domain.Entities
{
    public class Category
    {
        
        public Guid Id { get; set; }
        [Required]
        [StringLength(100)]
        public string Name { get; set; }

        [JsonIgnore]
        public ICollection<Topic> Topics { get; set; }
    }
}

File: forum-client/src/app/models/category.ts
export default  interface Category{
    id: string;
    name: string;
}

export class CategoryValues implements Category{
    id!: string;
    name!: string;

    constructor(category?: Category) {
        if (category) {
          this.id = category.id;
          this.name = category.name
        }
    }

}